apps/framework-cli/src/framework/python/parser.rs

```diff
@@ -0,0 +1,465 @@
+//! # Python Parser
+//! This module is responsible for parsing the python schema file and extracting the data model from it
+//!
+//! There are two main capabilities in this module:
+//! 1. Extracting python from a schema file and turning it an AST
+//! 2. Mapping that AST into file objects
+//!
+//! ## File Objects
+//! The file objects are all the data model objects that are extracted from the python schema file
+//! and it's associsted supporting objects such as enums.
+
+use std::path::PathBuf;
+
+use rustpython_parser::{
+    ast::{self, Expr, Identifier, Stmt},
+    Parse,
+};
+
+use crate::{
+    framework::schema::{Column, ColumnDefaults, ColumnType, DataEnum as FrameworkEnum, DataModel},
+    project::PROJECT,
+};
+
+#[derive(Debug, Clone, thiserror::Error)]
+#[error("Failed to parse the python file")]
+#[non_exhaustive]
+pub enum PythonParserError {
+    FileNotFound {
+        path: PathBuf,
+    },
+    #[error("Python Parser - Unsupported data type: {type_name}")]
```

Comment:
Here you will print `Python Parser - Unsupported data type: Unsupported data type` if you don't specify the data type properly at the Error building site